cpp/order/OrderParameter.h

```diff
@@ -0,0 +1,167 @@
+        [=] (size_t i, std::shared_ptr<freud::locality::NeighborIterator::PerPointIterator> ppiter)
+        {
+            m_psi_array.get()[i] = 0;
+            vec3<float> ref = (*points)[i];
+
+            for(freud::locality::NeighborBond nb = ppiter->next(); !ppiter->end(); nb = ppiter->next())
+            {
+                // Compute r between the two particles
+                vec3<float> delta = m_box.wrap((*points)[nb.ref_id] - ref);
+
+                // Compute psi for neighboring particle
+                // (only constructed for 2d)
+                m_psi_array.get()[i] += func(delta);
+            }
+
+            m_psi_array.get()[i] /= std::complex<float>(m_k);
+        });
+
+    //     freud::locality::loopOverNeighborsPoint(points, points->getRefPoints(), Np, qargs, nlist, 
+    //     [=](size_t i)
+    //     {
+    //         m_psi_array.get()[i] = 0; return 0;
+    //     }, 
+    //     [=](size_t i, size_t j, float distance, float weight, int data)
+    //     {
+    //         vec3<float> ref = points->getRefPoints()[i];
+    //         // Compute r between the two particles
+    //         vec3<float> delta = m_box.wrap(points->getRefPoints()[j] - ref);
+
+    //         // Compute psi for neighboring particle
+    //         // (only constructed for 2d)
+    //         m_psi_array.get()[i] += func(delta);
+    //     },
+    //     [=](size_t i, int data)
+    //     {
+    //         m_psi_array.get()[i] /= std::complex<float>(m_k);
+    //     });
+
+        // Save the last computed number of particles
+        m_Np = Np;
+    }
+
+    T getK()
+    {
+        return m_k;
+    }
+
+    unsigned int getNP()
+    {
+        return m_Np;
+    }
+
+
+protected:
+    box::Box m_box;    //!< Simulation box where the particles belong
+    unsigned int m_Np; //!< Last number of points computed
+    T m_k;
+    std::shared_ptr<std::complex<float>> m_psi_array; //!< psi array computed
+};
+
+//! Compute the translational order parameter for a set of points
+/*!
+ */
+class TransOrderParameter : public OrderParameter<float>
+{
+public:
+    //! Constructor
+    TransOrderParameter(float k = 6);
+
+    //! Destructor
+    ~TransOrderParameter();
+
+    //! Compute the translational order parameter
+    void compute(const freud::locality::NeighborList* nlist,
+                 const freud::locality::NeighborQuery* points, freud::locality::QueryArgs qargs);
+
+    //! Get a reference to the last computed dr
+    std::shared_ptr<std::complex<float>> getDr()
+    {
+        return m_psi_array;
+    }
+};
+
+//! Compute the hexagonal order parameter for a set of points
+/*!
+ */
+class HexOrderParameter : public OrderParameter<unsigned int>
+{
+public:
+    //! Constructor
+    HexOrderParameter(unsigned int k = 6);
+
+    //! Destructor
+    ~HexOrderParameter();
+
+        //! Get a reference to the last computed psi
+    std::shared_ptr<std::complex<float>> getPsi()
+    {
+        return m_psi_array;
+    }
```

Comment:
Move this getter up to the parent class; since we're using a common array, I would remove the getDr function in TransOrderParameter and just use this function. Maybe rename it to `getOrder` or so. We definitely don't need to preserve API compatibility on the C++ side, so you can have the current Cython functions call the parent instead.